### DISCUSSION TITLE
Export texture for model

### DISCUSSION THREAD

**Discussion by mirgorodalexander** [NVlabs/instant-ngp#397] [Q&A]
Hi, is there a way to get mesh with texture file with correct UV? For creating textured model.

**Comment by Tom94** [comment]
No: the mesh export currently only gives you per-face texturing / vertex colors. But you can import the mesh in third party software and unwrap it there. Cheers!

**Reply by benbeezy** [reply]
Do you have any suggestions for texturing the mesh?

**Reply by asetsuna** [reply]
xatlas is an option

**Reply by Furretar** [reply]
I figured it out after a few hours. You need to bake the embedded vertex colors to a texture in blender. First, you need to make it a Smart UV Project, doesn't work without it. (Enter edit mode, then at the top click UV, Smart UV Project)
![image](https://user-images.githubusercontent.com/101470861/205521972-ab1abb3c-a792-49b8-86b9-70e74bba9e43.png)

In the shading workspace you need to add the nodes image texture (where it sends the texture) and color attribute (another term for vertex color). You can add notes with the hotkey shift + a in the node editor. Connect Color from Color Attribute to Base Color on your shader (Principled BSDF by default). As far as I know every other setting in the shader node doesn't do anything in baking this. For the image texture node just click new and set a size and name. Here's my node layout:
![nodes2](https://user-images.githubusercontent.com/101470861/205521483-efd7117b-ed5a-41e8-9b5f-7f39a990a538.png)

For the bake settings, open the render properties tab.
![image](https://user-images.githubusercontent.com/101470861/205521524-4933e3fd-ac6d-43f8-99f7-1bb6efbdfb4a.png)

Set render engine to cycles. Under the Bake tab, set bake type to diffuse, and uncheck Direct and Indirect, leaving only Color. Make sure the Target is Image Textures, and click bake. Here are my bake settings:
![bakesettings](https://user-images.githubusercontent.com/101470861/205521594-1efdb2f0-50a2-46b4-b4c6-34d90c1c6232.png)

You can save your texture in the image editor window:
![image](https://user-images.githubusercontent.com/101470861/205521922-a9dc5c7d-e645-4969-9f6c-2c918d343cd0.png)

I hope I didn't miss anything, leave any questions if you have them.

**Reply by nepfaff** [reply]
I have not been very successful with this. I unwrapped the mesh before these steps using Blender unwrap (got an error otherwise). How did you unwrap the mesh?
